cpp/include/cudf/detail/aggregation/device_aggregators.cuh

```diff
@@ -181,6 +182,41 @@ struct update_target_element<
   }
 };
 
+template <typename Source>
+struct update_target_element<
+  Source,
+  aggregation::SUM_ANSI,
+  cuda::std::enable_if_t<std::is_same_v<Source, int64_t> && cudf::has_atomic_support<Source>()>> {
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index,
+                             column_device_view source,
+                             size_type source_index) const noexcept
+  {
+    using Target      = target_type_t<Source, aggregation::SUM_ANSI>;
+    auto source_value = static_cast<Target>(source.element<Source>(source_index));
+
+    // For SUM_ANSI, detect overflow using atomic compare-and-swap
+    Target* target_ptr = &target.element<Target>(target_index);
+    Target assumed;
+    Target new_val;
+    do {
+      assumed = *target_ptr;
+      // Check for overflow before addition
+      if ((source_value > 0 && assumed > INT64_MAX - source_value) ||
+          (source_value < 0 && assumed < INT64_MIN - source_value)) {
+        // Overflow detected - set target to null and return immediately
+        target.set_null(target_index);
+        return;
+      }
+      new_val = assumed + source_value;
+    } while (!cuda::atomic_ref<Target, cuda::thread_scope_device>(*target_ptr)
+                .compare_exchange_strong(assumed, new_val, cuda::std::memory_order_relaxed));
+
+    // Only set target to valid if no overflow occurred and target was previously null
+    if (target.is_null(target_index)) { target.set_valid(target_index); }
```

Comment:
I'm just noticing that all of these `update_target_element` functions are reading values from the target even when target row is marked as null/invalid. Seems we should not be including whatever garbage is at a null entry in the calculation. That is, we should check `target.is_null(target_index)` first and just return if `true`. No reason to read the source or do any calculation at all if the target row is null.